File: LiarBar-back/src/main/java/org/huex/liarbarback/models/Player.java
package org.huex.liarbarback.models;
import lombok.*;

@Getter @Setter @AllArgsConstructor
public class Player {
    private String userId;
    private String name;
    private boolean isActive;
    private boolean isReady;
    private boolean isHost;
    private String roomId;

    public Player(String userId) {
        this.userId = userId;
        this.name = "Player_" + userId.substring(0, 5); // Default name based on userId
        this.isActive = true;
        this.isReady = false;
        this.isHost = false; // Default to not being a host
    }

    @Override
    public String toString() {
        return userId.substring(0,5)+"\t"
            +name+"\t"
            +(isActive?"Active":"Inactive")+"\t"
            +(isReady?"Ready":"NotReady")+"\t"
            +(isHost?"Host":"NotHost")+"\n";
    }
}

File: LiarBar-front_unity20220322f1/Assets/Scripts/Models/Player.cs
using System;
using System.Collections;
using System.Collections.Generic;
using UnityEngine;

[Serializable]
public class Player
{
    public string userId;
    public string name;
    public bool active;
    public bool ready;
    public bool host;
    public bool roomId;

    public Player(string userId, string name, bool active, bool ready, bool host, bool roomId)  
    {
        this.userId = userId;
        this.name = name;
        this.active = active;
        this.ready = ready;
        this.host = host;
        this.roomId = roomId;
    }
    public Player() { }
    public override string ToString()
    {
        return JsonUtility.ToJson(this);
    }
}
